### PR TITLE
Prepare upgrade of dev qa prod

### PR DESCRIPTION
**Pull request by aliberkhsl**
#160

### CODE REVIEW

File: scripts/upgradeToV2/upgradeToV2.ts [hunk 1]
```diff
@@ -320,7 +320,10 @@ export const main = async (networkName = name) => {
 
     release["StakingContracts"] = StakingContracts;
     release["DonationsStaking"] = DonationsStaking;
-
+    release["Identity"] = dao.Identity;
+    release["GoodDollar"] = dao.GoodDollar;
+    release["Contribution"] = dao.Contribution;
+    release["Bridge"] = dao.Bridge;
```
Comment (sirpy) [inline scripts/upgradeToV2/upgradeToV2.ts:326]: there's no such field. its foreignbridge or homebridge.

File: scripts/upgradeToV2/upgradeToV2.ts [hunk 2]
```diff
@@ -320,7 +320,10 @@ export const main = async (networkName = name) => {
...
     console.log({ StakingContracts, DonationsStaking });
     let res = Object.assign(newdao, release);
     await releaser(release, networkName);
```
Comment (sirpy) [inline scripts/upgradeToV2/upgradeToV2.ts:329]: avatar and controller are missing
also all the fuse specific contracts are missing

**Review by sirpy — CHANGES_REQUESTED** [review]

File: scripts/upgradeToV2/upgradeToV2.ts (hunk 1 above)
Comment (aliberkhsl) [reply scripts/upgradeToV2/upgradeToV2.ts]: Changed

File: airdrop.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (sirpy) [inline airdrop.json:1]: this should be the test airdrop

File: releases/deploy-settings.json [hunk 4]
```diff
@@ -64,14 +64,14 @@
   },
   "fuse": {
     "governance": {
-      "gdaoAirdrop": "0x2bc5d079ee076bebadc354ad388aff81fb0ade0b3fbb663d5ce45ec63bec31a2",
+      "gdaoAirdrop": "0x9bf0fb0be13c4e1ab1da89850b23ab7919116fb9fc60ad08ec06bbca58516d11",
```
Comment (sirpy) [inline releases/deploy-settings.json:67]: what about staging env

File: releases/deployment.json [hunk 5]
```diff
@@ -60,5 +60,19 @@
       ["0xf5e4C09Cd8A01F066f976A0c17f88e549817ea17", 6944]
     ],
     "DonationsStaking": "0xe58501733711B966f9c1d81B29557674c11bD840"
+  },
+  "fuse": {
```
Comment (sirpy) [inline releases/deployment.json:64]: missing contracts
make sure you have everything from contracts,stakingmodel,upgradeables

File: reptree.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+[
```
Comment (sirpy) [inline reptree.json:1]: in test

File: releases/olddao.json [hunk 7]
```diff
@@ -107,5 +107,109 @@
     "DonationsStaking": "0xa30F00b4b2d02d1E02f48B645B04E60125dF6D49",
     "network": "kovan-mainnet",
     "networkId": 42
+  },
+  "fuse": {
```
Comment (sirpy) [inline releases/olddao.json:111]: missing contracts from fuse
where is staging?

**Review by sirpy — CHANGES_REQUESTED** [review]

File: releases/deploy-settings.json (hunk 4 above)
Comment (aliberkhsl) [reply releases/deploy-settings.json:67]: Replaced

File: releases/deployment.json [hunk 8]
```diff
@@ -73,6 +74,49 @@
...
+    "Avatar": "0x3BC91a65f53dACBd2336C04AC060741f70dCd187",
+    "Controller": "0x589F7ae100a7dbd80DC512208b7a0F457A041BE6",
+    "AbsoluteVote": "0x11e7716de4B5bC9A1dBc93184b27aC2134CA6E42",
+    "SchemeRegistrar": "0x4F151834efb18108080716d1467b397FAA87cF66",
+    "UpgradeScheme": "0xbd879865b649B7605c27cAa6A54e0f6B9Efb74b2",
+    "ForeignBridge": {
+      "_foreignBridge": "0xEDA39475415f1A2944a467Aa6359CB4C1c3ed50f",
+      "_foreignValidators": "0x9ebe7b1b0fB6E5D4c3996e65AB3750Fbbf8958aA",
+      "_foreignToken": "0x4738C5e91C4F809da21DD0Df4B5aD5f699878C1c",
+      "_blockNumber": 8259220
```
Comment (sirpy) [inline releases/deployment.json]: all this needs to go to olddao
in the upgrade code you merge/copy into the release object which is then written to deployment

File: scripts/upgradeToV2/upgradeToV2.ts [hunk 9]
```diff
@@ -556,6 +556,7 @@ export const main = async (networkName = name) => {
   };
 
   const deployStakingContracts = async release => {
+    const isRopsten = networkName !== "fuse-mainnet";
```
Comment (sirpy) [inline scripts/upgradeToV2/upgradeToV2.ts:559]: ropsten is fuse-mainnet and staging-mainnet

File: scripts/upgradeToV2/upgradeToV2.ts [hunk 10]
```diff
@@ -603,43 +604,47 @@ export const main = async (networkName = name) => {
...
-
-    const aaveps = aaveTokens.map(async token => {
-      let rewardsPerBlock = (protocolSettings.staking.rewardsPerBlock / 2) //aave gets half of the rewards
-        .toFixed(0);
-      console.log("deployStakingContracts", {
-        token,
-        settings: protocolSettings.staking,
-        rewardsPerBlock
+    let deployed;
+    if (isRopsten) {
```
Comment (sirpy) [inline scripts/upgradeToV2/upgradeToV2.ts:608]: if !isRopsten

**Review by sirpy — CHANGES_REQUESTED** [review]

File: scripts/upgradeToV2/upgradeToV2.ts (hunk 10 above)
Comment (aliberkhsl) [reply scripts/upgradeToV2/upgradeToV2.ts]: changed

File: scripts/upgradeToV2/upgradeToV2.ts (hunk 9 above)
Comment (aliberkhsl) [reply scripts/upgradeToV2/upgradeToV2.ts]: added

File: releases/deployment.json (hunk 8 above)
Comment (aliberkhsl) [reply releases/deployment.json]: moved

File: scripts/upgradeToV2/upgradeToV2.ts [hunk 11]
```diff
@@ -320,6 +322,18 @@ export const main = async (networkName = name) => {
...
+    release["Contribution"] = dao.Contribution;
+    release["ForeignBridge"] = dao.ForeignBridge;
+    release["HomeBridge"] = dao.HomeBridge;
+    release["Controller"] = dao.Controller;
+    release["Avatar"] = avatar;
+    release["FirstClaimPool"] = dao.FirstClaimPool;
+    release["Reputation"] = dao.Reputation;
+    release["AbsoluteVote"] = dao.AbsoluteVote;
+    release["SchemeRegistrar"] = dao.SchemeRegistrar;
+    release["UBI"] = dao.UBI;
```
Comment (sirpy) [inline scripts/upgradeToV2/upgradeToV2.ts:336]: add only contracts we DONT UPGRADE.
still not all non upgraded contracts are added
remove undefined such as homebridge on mainnet

**Review by sirpy — CHANGES_REQUESTED** [review]

File: hardhat.config.ts [hunk 12]
```diff
@@ -86,8 +86,8 @@ const hhconfig: HardhatUserConfig = {
     "fuse-mainnet": {
       accounts: { mnemonic },
       url: "https://ropsten.infura.io/v3/" + infura_api,
-      gas: 9000000,
-      gasPrice: 2500000000,
+      gasPrice: 20e9,
+      gas: 7e6,
```
[the remark this answers is not in the document]
Comment (aliberkhsl) [reply hardhat.config.ts:90]: I did

File: scripts/upgradeToV2/upgradeToV2.ts (hunk 11 above)
Comment (aliberkhsl) [reply scripts/upgradeToV2/upgradeToV2.ts]: Removed them

File: reptree.json (hunk 6 above)
Comment (aliberkhsl) [reply reptree.json]: changed

File: scripts/upgradeToV2/upgradeToV2.ts [hunk 13]
```diff
@@ -327,6 +327,7 @@ export const main = async (networkName = name) => {
     } else {
       release["ForeignBridge"] = dao.ForeignBridge;
       release["Contribution"] = dao.Contribution;
+      release["AdminWallet"] = dao.AdminWallet;
```
Comment (sirpy) [inline scripts/upgradeToV2/upgradeToV2.ts:330]: adminwallet is not on mainnet

**Review by sirpy — CHANGES_REQUESTED** [review]

File: scripts/upgradeToV2/upgradeToV2.ts (hunk 13 above)
Comment (aliberkhsl) [reply scripts/upgradeToV2/upgradeToV2.ts]: Changed

File: releases/deploy-settings.json [hunk 14]
```diff
@@ -64,14 +64,15 @@
...
       "gdaoTotalSupply": "24000000000000000000000000",
       "proposalVotingPeriod": 241920
     }
   },
   "fuse-mainnet": {
+    "gdxAirdrop": "0x87db9fa52d55498bbf1e6005603afe05cb194c13e8ac4907282f4ea2717d0921",
```
[the remark this answers is not in the document]
Comment (aliberkhsl) [reply releases/deploy-settings.json]: Replaced gdxairdrop hash for new mnemonic

**Pull request merged**